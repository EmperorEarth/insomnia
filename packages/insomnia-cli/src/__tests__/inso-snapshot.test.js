// @flow
import execa from 'execa-wrap';

// MAKE SURE YOU BUILD THE PROJECT BEFORE RUNNING THESE TESTS.
// These tests use the executable /bin/inso, which relies on /dist.
describe('bin/inso snapshots', () => {
  describe('base', () => {
    it.each([
      '-h',
      '--help',
      'help',
      'generate -h',
      'generate config -h',
    ])('shows help page with "%s"', async args =>
      expect(await execa('bin/inso', args.split(' '))).toMatchSnapshot(),
    );
  });
});
