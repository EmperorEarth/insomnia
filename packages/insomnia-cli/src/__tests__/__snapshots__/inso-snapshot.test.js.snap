// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin/inso snapshots base shows help page with "--help" 1`] = `
"
  command: bin/inso --help
  code: 0
  failed: false
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso [options] [command]

  A CLI for Insomnia!

  Options:
    -v, --version   output the version number
    -h, --help      display help for command

  Commands:
    generate        Code generation utilities
    help [command]  display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;

exports[`bin/inso snapshots base shows help page with "-h" 1`] = `
"
  command: bin/inso -h
  code: 0
  failed: false
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso [options] [command]

  A CLI for Insomnia!

  Options:
    -v, --version   output the version number
    -h, --help      display help for command

  Commands:
    generate        Code generation utilities
    help [command]  display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;

exports[`bin/inso snapshots base shows help page with "generate -h" 1`] = `
"
  command: bin/inso generate -h
  code: 0
  failed: false
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso generate [options] [command]

  Code generation utilities

  Options:
    -h, --help                   display help for command

  Commands:
    config [options] <filePath>  Generate configuration from an api spec
    help [command]               display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;

exports[`bin/inso snapshots base shows help page with "generate config -h" 1`] = `
"
  command: bin/inso generate config -h
  code: 0
  failed: false
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso generate config [options] <filePath>

  Generate configuration from an api spec

  Options:
    -t, --type <value>   the type of configuration to generate, options are
                         [kubernetes, declarative]
    -o, --output <path>  the output path
    -h, --help           display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;

exports[`bin/inso snapshots base shows help page with "help" 1`] = `
"
  command: bin/inso help
  code: 0
  failed: false
  killed: false
  signal: null
  timedOut: false

  stdout:
  -------
  Usage: inso [options] [command]

  A CLI for Insomnia!

  Options:
    -v, --version   output the version number
    -h, --help      display help for command

  Commands:
    generate        Code generation utilities
    help [command]  display help for command
  -------
  stderr:
  -------
  
  -------
  "
`;
